scilpy/tractanalysis/mrds_along_streamlines.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+from scilpy.tractanalysis.grid_intersections import grid_intersections
+
+
+def mrds_metrics_along_streamlines(sft, mrds_pdds,
+                                   metrics, max_theta,
+                                   length_weighting):
+    """
+    Compute mean map for a given lobe-specific metric along streamlines.
+
+    Parameters
+    ----------
+    sft : StatefulTractogram
+        StatefulTractogram containing the streamlines needed.
+    mrds_pdds : ndarray (X, Y, Z, 3*N_TENSORS)
+        MRDS principal diffusion directions of the tensors
+    metrics : list of ndarray
+        Array of shape (X, Y, Z, N_TENSORS) containing the fixel-specific
+        metric of interest.
+    max_theta : float
+        Maximum angle in degrees between the fiber direction and the
+        MRDS principal diffusion direction.
+    length_weighting : bool
+        If True, will weigh the metric values according to segment lengths.
+    """
+
+    mrds_sum, weights = \
+        mrds_metric_sums_along_streamlines(sft, mrds_pdds,
+                                           metrics, max_theta,
+                                           length_weighting)
+
+    all_metric = mrds_sum[0]
+    for curr_metric in mrds_sum[1:]:
+        all_metric += curr_metric
+
+    non_zeros = np.nonzero(all_metric)
+    weights_nz = weights[non_zeros]
+    for metric_idx in range(len(metrics)):
+        mrds_sum[metric_idx][non_zeros] /= weights_nz
+
+    return mrds_sum
+
+
+def mrds_metric_sums_along_streamlines(sft, mrds_pdds, metrics,
+                                       max_theta, length_weighting):
+    """
+    Compute a sum map along a bundle for a given fixel-specific metric.
+
+    Parameters
+    ----------
+    sft : StatefulTractogram
+        StatefulTractogram containing the streamlines needed.
+    mrds_pdds : ndarray (X, Y, Z, 3*N_TENSORS)
+        MRDS principal diffusion directions (PDDs) of the tensors
+    metrics : list of ndarray (X, Y, Z, N_TENSORS)
+        The fixel-specific metric of interest.
+    max_theta : float
+        Maximum angle in degrees between the fiber direction and the
+        MRDS principal diffusion direction.
+    length_weighting : bool
+        If True, will weight the metric values according to segment lengths.
+
+    Returns
+    -------
+    metric_sum_map : np.array
+        fixel-specific metric sum map.
+    weight_map : np.array
+        Segment lengths.
+    """
+
+    sft.to_vox()
+    sft.to_corner()
+
+    X, Y, Z = metrics[0].shape[0:3]
+    metrics_sum_map = np.zeros((len(metrics), X, Y, Z))
+    weight_map = np.zeros(metrics[0].shape[:-1])
+    min_cos_theta = np.cos(np.radians(max_theta))
+
+    all_crossed_indices = grid_intersections(sft.streamlines)
```

Comment:
I am pretty sure that our "uncompress" method should be used somewhere here. @frheault probably knows this better than me.